packages/near-fast-auth-signer-e2e-tests/utils/firebase.ts

```diff
@@ -12,25 +14,103 @@ import PageManager from '../pages/PageManager';
 
 const FIREBASE_API_KEY_TESTNET = 'AIzaSyDAh6lSSkEbpRekkGYdDM5jazV6IQnIZFU';
 
+type MockAccount = {
+  type: 'email'|'uid',
+  email?: string,
+  uid?: string
+}
+
+function getRandomWaitTime(min, max) {
+  // Generate a random number between min and max (inclusive)
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+const performTaskWithRandomDelay = async () => {
+  const randomWaitTime = getRandomWaitTime(1000, 5000);
+
+  await new Promise((resolve) => { setTimeout(resolve, randomWaitTime); });
+};
+
+export const isServiceAccountAvailable = () => {
+  if (
+    !serviceAccount.private_key_id
+    || !serviceAccount.private_key
+    || !serviceAccount.client_email
+    || !serviceAccount.client_id
+    || !serviceAccount.client_x509_cert_url
+  ) {
+    return false;
+  }
+  return true;
+};
+
 export const initializeAdmin = () => {
-  admin.initializeApp({
-    credential: admin.credential.cert(serviceAccount),
-  });
+  if (isServiceAccountAvailable() && !admin.apps.length) {
+    admin.initializeApp({
+      credential: admin.credential.cert(serviceAccount)
+    });
+  }
 };
 
 export const deleteAccount = async (userUid: string) => {
-  if (userUid) {
-    const user = await admin.auth().getUser(userUid);
+  try {
+    const user = userUid && await admin.auth().getUser(userUid);
     if (user.uid) {
+      // delete user auth record
       await admin.auth().deleteUser(userUid);
     }
+  } catch (error) {
+    if (error.code !== 'auth/user-not-found') {
+      console.error('Error deleting user auth record:', error);
+    }
+  }
+
+  try {
+    const db = admin.firestore();
+    const docRef = db.collection('users').doc(userUid);
+    // delete user collection record
+    await docRef.delete();
+  } catch (error) {
+    console.error('Error deleting user Firestore record:', error);
+  }
+};
+
+export const deleteUserByEmail = async (email: string) => {
+  try {
+    const user = await admin.auth().getUserByEmail(email);
+    if (user && user.uid) {
+      await deleteAccount(user.uid);
+    }
+  } catch (error) {
+    if (error.code !== 'auth/user-not-found') {
+      console.error('Error fetching user by email:', error);
+    }
   }
 };
 
-const addAccountPublicKeyToFirestore = async (accountId: string, publicKey: string) => {
-  const docRef = admin.firestore().collection('publicKeys').doc(publicKey);
+export const deletePublicKey = (publicKey: string) => {
+  const db = admin.firestore();
+  const docRef = db.collection('publicKeys').doc(publicKey);
+  return docRef.delete();
+};
 
-  await docRef.set({ accountId });
+export const addAccountToBeDeleted = async (account: MockAccount) => {
```

Comment:
I/O on parallel test can have race condition issues, no? 

I think we need a locker to make sure we are not doing I/O in parallel.